okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingConcurrencyLimiter.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * TODO(jbaker): Add Netflix copyright license once they add it to their code.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.okhttp;
+
+import com.netflix.concurrency.limits.Limit;
+import com.netflix.concurrency.limits.Limiter;
+import com.netflix.concurrency.limits.Strategy;
+import com.netflix.concurrency.limits.Strategy.Token;
+import com.netflix.concurrency.limits.limit.VegasLimit;
+import com.netflix.concurrency.limits.limiter.ImmutableSampleWindow;
+import com.netflix.concurrency.limits.strategy.SimpleStrategy;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.LongSupplier;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * This class is heavily based on Netflix's {@link com.netflix.concurrency.limits.limiter.DefaultLimiter}, and omits
+ * the license header because that class, while claimed in the repository to be Apache 2.0 licensed, itself contains
+ * <a href="https://github.com/Netflix/concurrency-limits/issues/73">no license header</a>.
+ * <p>
+ * Changes are as follows:
+ * <ol>
+ * <li>Fix concurrency bug - upstream has a race between starting finishing the window and running requests.</li>
+ * <li>Store the last update time as well as the next update time.</li>
+ * <li>Use the last update time to pre-emptively close the window when a request is dropped, instead of waiting.
+ *     Do this before handing back any tokens.</li>
+ * <li>Remove the builder, and instead have a static factory method.</li>
+ * <li>Change code style to match Palantir baseline.</li>
+ * </ol>
+ */
+final class RemotingConcurrencyLimiter implements Limiter<Void> {
```

Comment:
Not sure how to review this. Are the things listed above expected to merge upstream so we can remove this class at some point?